WireDomain/Sources/WireDomain/Utilities/Journal/JournalStore.swift

```diff
@@ -0,0 +1,78 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// An object that facilitates access to the persistent
+/// journal entries for a particular user.
+
+public struct JournalStore {
```

Comment:
This is intended as a general purpose tool to help in with interruptible flows. For instance, considering a particular migration, a single journal entry could keep track of several flags indicating if the migration is required, whether it is has begun, and whether it was completed.